packages/mobile/src/identity/commentEncryption.ts

```diff
@@ -225,38 +224,39 @@ function* verifyIdentityMetadata(data: IdentityMetadataInTx[]) {
     return []
   }
 
-  const phoneHashes = new Set<string>()
-  data.map((d) => {
-    const phoneHash = getPhoneHash(d.e164Number, d.salt)
-    phoneHashes.add(phoneHash)
-    d.phoneHash = phoneHash
-  })
+  const verifiedTx = []
 
-  const lookupResult: IdentifierLookupResult = yield call(
-    lookupAttestationIdentifiers,
-    Array.from(phoneHashes)
-  )
+  for (const metadata of data) {
+    const phoneHash = getPhoneHash(metadata.e164Number, metadata.salt)
+    metadata.phoneHash = phoneHash
+    const lookupResult: string[] = yield call(lookupAttestationIdentifiers, phoneHash)
 
-  return data.filter((d) => {
-    const onChainAddresses = getAddressesFromLookupResult(lookupResult, d.phoneHash!)
+    // Check that there are verified addresses.
+    const onChainAddresses: string[] = yield call(
+      getAddressesFromLookupResult,
+      lookupResult,
+      phoneHash
+    )
     if (!onChainAddresses || !onChainAddresses.length) {
       Logger.warn(
         TAG + 'verifyIdentityMetadata',
-        `Phone number and/or salt claimed by address ${d.address} is not verified. Values are incorrect or sender is impersonating another number`
+        `Phone number and/or salt claimed by address ${metadata.address} is not verified. Values are incorrect or sender is impersonating another number`
       )
-      return false
+      continue
     }
 
-    if (!onChainAddresses.find((a) => eqAddress(a, d.address))) {
+    if (!onChainAddresses.find((a) => eqAddress(a, metadata.address))) {
```

Comment:
metadata.address is the wallet or account address? IIUC it should be the account address. Should we then be checking that the wallet address for that account address was the sender of the tx?

I guess I don't see how this verifies the transaction, probably I'm missing something somewhere. What's stopping me from sending a tx that I claim is from someone else?

Reply:
Yeah, there's something I'm missing here. The comments on payments are shown fine and I see the phone number as the sender but this equality fails and I see the message below in the logs. This method is used to update phone numbers to salt and phone number to address mappings. I'm not sure how those are used, I guess it makes some of our caches useless? I feel like even if we fix this here it will have repercussions in other places that I don't fully understand.

In any case we'll need to get the wallet address here, I guess that requires a call to the accounts contract? I'll check if we're requesting it somewhere else.

Reply:
It's the wallet address. We basically want to do what you suggested but in reverse.
1. Compute `phoneHash` from the phone number and salt included in the metatx
2. Use `phoneHash` to lookup the `identifiers`
3. Get the `verifiedAccounts` associated with the `identifiers`
4. Get the `walletAddresses` registered to those `verifiedAccounts`
5. Confirm that at least one of the `walletAddresses` is equal to the address the transaction was sent from

I think we are missing step 4 right now.

Reply:
Alternatively, we could also use the `lookupIdentifiers` as that returns the `walletAddresses` straight away, though there is a bit of trade-off with the logic not being explicit as we've seen

Reply:
@tarikbellamine +1 to your first suggestion

Reply:
I think view calls for now but there should be no reason we need to look up this info more than once for the vast majority of cases. I'm hesitant to throw more caches in the mix because, as you've noted, our caching situation right now is kinda messy and should be better organized.

If you feel strongly about the performance hit, it's probably something we could tack on. Otherwise, it may be better to wait until we take a more comprehensive look at how and when we are caching across Valora. Will leave the call up to you and @gnardini

Reply:
This particular function is not called very often, so I'm nor sure it would add much value. I do think that having a unified cache like @tarikbellamine suggested would work best, but we have to rethink our caches app-wide which we could hopefully tackle soon.

Reply:
Definitely do not feel strongly about the performance hit and agree with both of you that this is the sort of thing better left to a refactor, just wanted to flag